### ISSUE TITLE
[incubator/druid] druid is not  installing with default chart value on AWS EKS (k8S) cluster

### ISSUE THREAD

**Issue by tmohanta** [helm/charts#14725]
**Describe the bug**
The helm installation  of the incubator/druid with the default chart value is not working on AWS EKS cluster. The druid related pods are inifinitely  going on the CrashLoopBackOff state.
$ helm ls
NAME 	REVISION	UPDATED                 	STATUS  	CHART      	APP VERSION	NAMESPACE
druid	1       	Tue Jun 11 22:31:06 2019	DEPLOYED	druid-0.1.0	0.12.3     	druid   

**Version of Helm and Kubernetes**:
$ helm version
Client: &version.Version{SemVer:"v2.14.0", GitCommit:"05811b84a3f93603dd6c2fcfe57944dfa7ab7fd0", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.14.0", GitCommit:"05811b84a3f93603dd6c2fcfe57944dfa7ab7fd0", GitTreeState:"clean"}
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.2", GitCommit:"66049e3b21efe110454d67df4fa62b08ea79a19b", GitTreeState:"clean", BuildDate:"2019-05-16T16:23:09Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.6-eks-d69f1b", GitCommit:"d69f1bf3669bf00b7f4a758e978e0e7a1e3a68f7", GitTreeState:"clean", BuildDate:"2019-02-28T20:26:10Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}

**Which chart**:
incubator/druid

**What happened**:
Trying to install the druid using helm on the k8s cluster of aws (3 nodes).
The pods are not coming up. The state of the pods are CrashLoopBackOff.
$ kubectl get pods -n druid
NAME                                 READY   STATUS             RESTARTS   AGE
druid-broker-74769d575f-77lgc        0/1     CrashLoopBackOff   124        12h
druid-coordinator-5c49599bdb-wbd8k   0/1     Running            166        12h
druid-historical-0                   1/1     Running            103        12h
druid-middle-manager-0               0/1     Running            124        12h
druid-mysql-b8f6777f9-8flmx          1/1     Running            0          12h
druid-overlord-577d8d89fb-sp8h9      0/1     Running            164        12h
druid-zookeeper-0                    1/1     Running            6          11h
druid-zookeeper-1                    1/1     Running            0          12h
druid-zookeeper-2                    1/1     Running            0          12h

**What you expected to happen**:
All the druid related pods should be in the ready state.

**How to reproduce it** (as minimally and precisely as possible):
Install the druid from helm chart using the following command

helm install --namespace "druid" --name "druid" incubator/druid

**Anything else we need to know**:
Same issue facing while installing the incubator/kafka on the cluster. kafka pod is not coming to ready state although the zookeeper pods are in ready state.

**Comment by dwinters-gpsw** [comment]
We are seeing the same behavior where the broker and historical server are infinitely going to the CrashLoopBackOff status.  We are also running on EKS.

**Comment by ilhanadiyaman** [comment]
I also had same problems. In order to fix this issue I did steps below:
- If your nodes are running different regions, use [topology-aware storage class](https://kubernetes.io/blog/2018/10/11/topology-aware-volume-provisioning-in-kubernetes/) instead of default("gp2"). With default settings, EKS creates EBS storages in different AZ then node placed. Setup topology-aware-storage-class and change the values of the pods.

- Historical and broker pod request 8gb direct memory for JVM. If your node doesn't have enough available memory then it gives CrashLoopBackOff error. For testing purpose, I added an instance with 8 gb memory and changed historical pod's MaxDirectMemorySize to 6gb. It worked on my m5.large nodes. Probably, there should be at least two m5.xlarge nodes(except zk nodes) in order to setup everything without a headache.
